src/shared/components/Animations/InfiniteScroll.stories.tsx

```diff
@@ -0,0 +1,359 @@
+import * as React from 'react';
+import { storiesOf } from '@storybook/react';
+import { withInfo } from '@storybook/addon-info';
+import {
+  withKnobs,
+  text,
+  array,
+  object,
+  boolean,
+  number,
+} from '@storybook/addon-knobs';
+import { action } from '@storybook/addon-actions';
+import InfiniteScroll from './InfiniteScroll';
+import ListItem from './ListItem';
+import { FetchableState } from '../../store/reducers/utils';
+import { PaginatedList } from '@bbp/nexus-sdk';
+
+// @ts-ignore
+const shuffleArray = (arr: any[]) =>
+  arr
+    .map((a: any) => [Math.random(), a])
+    .sort((a: any, b: any) => a[0] - b[0])
+    .map((a: any) => a[1]);
+
+const exampleItems = [
+  'https://bbp-nexus.epfl.ch/vocabs/bbp/neurosciencegraph/core/v0.1.0/InVitroSliceReconstructedPatchedNeuron',
+  'https://bbp-nexus.epfl.ch/vocabs/bbp/neurosciencegraph/core/v0.1.0/ReconstructedCell',
+  'http://www.w3.org/ns/prov#Entity',
+  'https://bbp-nexus.epfl.ch/vocabs/bbp/neurosciencegraph/core/v0.1.0/ReconstructedPatchedCell',
+  'http://www.w3.org/ns/dcat#Dataset',
+];
+
+const mockServer = (numberOfPages: number, pageSize: number) => {
+  let position = 0;
+  let items = exampleItems;
+  return {
+    next: () => {
+      if (position >= numberOfPages) {
+        return;
+      }
+      position = position + 1;
+      items = shuffleArray(exampleItems);
+    },
+    getItems: () => items,
+  };
+};
+
+storiesOf('Components/InfiniteScroll/Basic', module)
+  .addDecorator(withKnobs)
+  .add(
+    'InfiniteScroll',
+    withInfo(`
+    Scroll all the way down and ask for more!
+
+    ~~~js
+    <InfiniteScroll
```

Comment:
I was expecting only 3 props:
- `items` a list of items to render
- `onItemClick(itemID)` callback when an item is clicked
- `onLoadMore()?` optional callback when load more button clicked (if none, no loadmore button)

Reply:
The items are agnostic basically so you can pass whatever event handlers you want from the parent

I also though to use the paginated settings conventions we used elsewhere in the app for this. 

This element takes care of handling the state of results, so that each page is concatenated to each other, so the parent doesn't need to do that

Reply:
What I don't get is that there is no pagination in this component, and yet we have to deal with  a `Paginated` list...

Reply:
Well there is pagination for sure, it's just that it's always sequential.